### PR TITLE
Test Cleanup 02

### PR DESCRIPTION
**Pull request by nathanwoctopusdeploy**
# Background

Prior to Async Halibut we are performing some test cleanup.

Change some tests to use ClientServiceBuilder

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Halibut.Tests/BackwardsCompatibility/InvocationExceptionsAreMappedCorrectlyFixture.cs [hunk 1]
```diff
@@ -10,28 +12,11 @@ namespace Halibut.Tests.BackwardsCompatibility
...
+        public async Task OldInvocationExceptionMessages_AreMappedTo_ServiceInvocationHalibutClientException(ServiceConnectionType serviceConnectionType)
         {
             using (var clientAndService = await ClientAndPreviousServiceVersionBuilder
-                       .WithPollingService()
-                       .WithServiceVersion(PreviousServiceVersions.v5_0_429)
-                       .Build())
-            {
-                var echo = clientAndService.CreateClient<IEchoService>(se =>
-                {
-                    se.PollingRequestQueueTimeout = TimeSpan.FromSeconds(20);
```
Comment (LukeButters) [inline source/Halibut.Tests/BackwardsCompatibility/InvocationExceptionsAreMappedCorrectlyFixture.cs:22]: What happened to these timeouts?

Comment (nathanwoctopusdeploy) [reply source/Halibut.Tests/BackwardsCompatibility/InvocationExceptionsAreMappedCorrectlyFixture.cs:22]: It's still there, it's a kind of nasty diff to read
image

Comment (nathanwoctopusdeploy) [reply source/Halibut.Tests/BackwardsCompatibility/InvocationExceptionsAreMappedCorrectlyFixture.cs:22]: 2 tests become 1 by using the test case source

**Comment by nathanwoctopusdeploy** [comment]
[sc-53136]

File: source/Halibut.Tests/Support/Reference.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿namespace Halibut.Tests.Support
+{
+    public class Reference<T>
```
Comment (LukeButters) [inline source/Halibut.Tests/Support/Reference.cs:3]: lol

**Review by LukeButters — APPROVED** [review]

File: source/Halibut.Tests/Support/Reference.cs (hunk 2 above)
Comment (nathanwoctopusdeploy) [reply source/Halibut.Tests/Support/Reference.cs:3]: It will live on forever 😆

**Pull request merged**